peer_mentoring/groups/templates/groups_index.html

```diff
@@ -10,13 +10,15 @@
         <div class="fw-bold"><a href="{% url 'groups:group_detail' group.id %}">{{ group }}</a></div>
         <p class="card-text">{{group.description}}</p>
       </div>
+      {% if request.user not in groups.members %}
       <div>
-        <form method="post" action="{% url 'groups:group_index' %}" style="display:inline;">
+        <form method="post" action="{% url 'groups:group_home' %}" style="display:inline;">
```

Comment:
bootstrap should have a class for inline forms, so you don't need this inline style